heroku_applink/addons/heroku_applink.py

```diff
@@ -0,0 +1,121 @@
+import asyncio
+import urllib.parse
+from typing import TypedDict, Optional
+from heroku_applink.utils.http_request import HttpRequestUtil
+from heroku_applink.context import ClientContext, Org, User, DataAPI
+from heroku_applink.utils.addon_config import (
+    resolve_addon_config_by_attachment_or_color,
+    resolve_addon_config_by_url,
+)
+import os
+
+http_request_util = HttpRequestUtil()
+
+class RequestOptions(TypedDict):
+    method: str
+    headers: dict[str, str]
+
+# How long to wait for the add-on call before giving up (in seconds)
+try:
+    HTTP_TIMEOUT_SECONDS = float(os.getenv("HEROKU_APPLINK_TIMEOUT", "5.0"))
+except ValueError:
+    HTTP_TIMEOUT_SECONDS = 5.0
+
+# Fields we expect in a successful authorization payload
+_REQUIRED_FIELDS = {
+    "org_id",
+    "org_domain_url",
+    "user_id",
+    "username",
+    "request_id",
+    "access_token",
+    "api_version",
+    "namespace",
+}
+
+
+async def get_authorization(
+    developer_name: str,
+    attachment_or_url: Optional[str] = None,
+) -> ClientContext:
+    """
+    Fetch authorization for a given Heroku AppLink developer.
+    Uses GET {apiUrl}/invocations/authorization?org_name=developer_name
```

Comment:
This is different from what's in the [spec](https://docs.google.com/document/d/1duFS1OJ_vRww4h_PZl0iCLjcFdx1qmIuvQCPol3PgJM/edit?tab=t.d4ogbhifno4m#heading=h.jbokm910jur3) 
```suggestion
    Uses GET {apiUrl}/invocations/authorizations/{developer_name}
```